zebrad/tests/acceptance.rs

```diff
@@ -383,14 +437,61 @@ fn misconfigured_ephemeral_mode() -> Result<()> {
     // Make sure the command was killed
     output.assert_was_killed()?;
 
-    // Check that ephemeral takes precedence over cache_dir
+    let expected_run_dir_file_names = match cache_dir_check {
+        // we created the state directory, so it should still exist
+        EphemeralCheck::ExistingDirectory => {
+            assert_with_context!(
+                ignored_cache_dir
+                    .read_dir()
+                    .expect("ignored_cache_dir should still exist")
+                    .count()
+                    == 0,
+                &output,
+                "unexpected items in ignored_cache_dir for ephemeral {:?} {:?}: {:?}",
+                cache_dir_config,
+                cache_dir_check,
+                ignored_cache_dir.read_dir().unwrap().collect::<Vec<_>>()
+            );
+
+            ["state", "zebrad.toml"].iter()
```

Comment:
fwiw you could probably do this in both branch arms and have it work by resolving both branch arms to `&'static [&'static str]` implicitly. Static promotion in rust is extremely cool.

```suggestion
            &["state", "zebrad.toml"]
```

Reply:
That's the first thing I tried, but it doesn't actually work, because the arrays are different sizes:
```
error[E0308]: `match` arms have incompatible types
   --> zebrad/tests/acceptance.rs:474:13
    |
440 |       let expected_run_dir_file_names = match cache_dir_check {
    |  _______________________________________-
441 | |         // we created the state directory, so it should still exist
442 | |         EphemeralCheck::ExistingDirectory => {
443 | |             assert_with_context!(
...   |
456 | |             &["state", "zebrad.toml"]
    | |             ------------------------- this is found to be of type `&[&str; 2]`
...   |
474 | |             &["zebrad.toml"]
    | |             ^^^^^^^^^^^^^^^^ expected an array with a fixed size of 2 elements, found one with 1 element
475 | |         }
476 | |     };
    | |_____- `match` arms have incompatible types
    |
    = note:   expected type `&[&str; 2]`
            found reference `&[&str; 1]`
```